### ISSUE TITLE
Nesting circuits causes crash - Optifine issue?

### ISSUE THREAD

**Issue by Hirsu** [amadornes/SCM-Localization-Issues#259]
I made a circuit, picked it up and tried placing it on another tile. As soon as I hover over the tile, all weird kinds of rendering occur (I'll insert a screenshot asap) and the small circuit itself doesn't get rendered. Actually placing the circuit down on the tile causes the client to crash, producing this [crash log](http://pastebin.com/Nxr0xjrV).

I took a look around your issues list and found a similar issue ([right here](https://github.com/amadornes/SCM-Localization-Issues/issues/110)) which led me to believe this might be an issue between SCM and optifine. Sure enough, after removing optifine the small circuit renders just fine on other tiles and placing it down produces no crashes.

I'm using the latest All The Mods modpack, which contains SCM 1.0.4, to which I've added the latest Optifine (1.10.2 HD U D4).

is there any way this issue can be resolved? For the time being I can work around this by using multitile circuits or by disabling optifine, but I'd rather use optifine and retain the circuit nesting functionality ;)

Please let me know if you'd like me to provide additional information.

Thanks.

**Comment by LiamJarvis** [comment]
from what i've heard is that optifine does some weird things with rendering and it can mess with how mods render block, items, etc. so to fix your problem you will have to remove optifine.

Atleast thats what ive been told im not 100% sure how optifine works

**Comment by Hirsu** [comment]
Yes, removing Optifine seems to make the problem go away. However, the ideal situation would be that both mods work nicely together. Removing either mod because it doesn't play nice with another always seems like a workaround rather than an actual solution. 

In the previous (and similar) issue I've mentioned in my opening post, the developer of Optifine mentions the issue is caused by SCM trying to use OpenGL functions on the chunk building thread. My understanding of his technical explanation is limited, but from what I understand this isn't caused by Optifine.

In the meantime, however, I've found a workaround. Apparently, only premade circuits (the ones created by using blueprints) cause rendering issues/crashes when nesting. Manually built circuits don't seem to have this issue. So for now I just create the circuits myself based off the blueprint and then place those on other circuits :)

**Comment by TheOldOne822** [comment]
It crashes when a "chip" is rendered with words on it so if you used a blueprint placed it down then shift right click with the screwdriver to hide it it will crash too. It is "Fixable" by removing optifine but that makes Minecraft unplayable on my system for some reason.
